lib/package.py

```diff
@@ -136,9 +136,11 @@ def _load(self):
         self.name = self.package_data.get('name')
         self.sources = self.package_data.get('sources', [])
         for source in self.sources:
+            source_name = source.keys()[0]
             if 'archive' not in source.values()[0]:
-                source_name = source.keys()[0]
                 source[source_name]['archive'] = self.name
+            if 'ref_to_fetch' not in source.values()[0] and source_name == 'git':
+                source[source_name]['ref_to_fetch'] = None
```

Comment:
I think its a little more readable to write:
```
if source_name == 'git':
    source[source_name].setdefault('ref_to_fetch', None)
```
The effect is the same as your code.
The same could be applied to the "archive" above, removing the `if`.